framework/include/utils/VectorCompositeFunctor.h

```diff
@@ -76,17 +75,19 @@ class VectorCompositeFunctor : public Moose::FunctorBase<VectorValue<T>>
         _x_comp(elem_side_qp, state), _y_comp(elem_side_qp, state), _z_comp(elem_side_qp, state)};
   }
 
+  using Moose::FunctorBase<VectorValue<T>>::evaluateGradient;
+  GradientType evaluateGradient(const ElemFromFaceArg & elem_from_face,
+                                unsigned int state) const override
+  {
+    return {_x_comp.gradient(elem_from_face, state),
```

Comment:
We have to specify the return type of the functor. In this case I've chosen `VectorValue<T>` hence the name. Since `VectorValue` always has `LIBMESH_DIM` components, we must have that many functors (or else make all the methods have to switch based off the number of user-supplied functors). I can add more constructors that take less functor parameters, and then fill the remaining slots with constant-0 functors. If you want a return type other than `VectorValue` that is more flexible, than we can create another functor. But I like having a Euclidean vector composite functor